src/Appwrite/Platform/Workers/Builds.php

```diff
@@ -401,7 +404,11 @@ protected function buildDeployment(Device $deviceForFunctions, Webhook $queueFor
                 }
 
                 $directorySize = $localDevice->getDirectorySize($tmpDirectory);
-                $functionsSizeLimit = (int)System::getEnv('_APP_FUNCTIONS_SIZE_LIMIT', '30000000');
+
+                $functionsSizeLimit = empty($plan['functionSize'])
+                    ? (int) System::getEnv('_APP_FUNCTIONS_SIZE_LIMIT', 30000000)
+                    : (int) $plan['functionSize'] * 1000 * 1000;
+
```

Comment:
Same as before. Also notice here you did `empty`, but above `isset`. Let's stay consistent - same implementation in both places.

Reply:
`int` casting was inconsistent in the existing code as well as in one case, we use a validator